docs/how-to/python_api.md

```diff
@@ -0,0 +1,188 @@
+in the sections below
+
+#### Create new Spark service accounts
+
+New Spark service accounts can be created by first creating a `ServiceAccount`
+domain object, and optionally specifying extra-properties, e.g. 
+
+```python
+from spark_client.domain import PropertyFile, ServiceAccount
+
+configurations = PropertyFile({"my-key": "my-value"})
+service_account = ServiceAccount(
+    name="my-spark",
+    namespace="default",
+    api_server=kube_interface.api_server,
+    primary=False,
+    extra_confs=configurations,
+)
+```
+
+The account can then be created using the registry
+
+```python
+service_account_id = registry.create(service_account)
+```
+
+This returns an id, which is effectively the `{namespace}:{username}`, e.g. "default:my-spark".
+
+#### Listing spark service accounts
+
+Once Spark service accounts have been created, these can be listed via
+
+```python
+spark_service_accounts = registry.all()
+```
+
+or retrieved using their ids
+
+```python
+retrieved_account = registry.get(service_account_id)
+```
+
+#### Delete service account
+
+The registry can also be used to delete existing service accounts
+
+```python
+registry.delete(primary_account_id)
+```
+
+or using an already existing `ServiceAccount` object:
+
+```python
+registry.delete(service_account.id)
+```
+
+#### Manage Primary Accounts
+
+In the Spark Client Snap, there is the notion of the primary service account, the 
+one that would be chosed by default, if no specific information is provided. The
+primary Spark service account can be set using 
+
+```python
+registry.set_primary(service_account_id)
+```
+
+or using an already existing `ServiceAccount` object:
+
+```python
+registry.set_primary(service_account.id)
+```
+
+The primary Spark service account can be retrieve using 
+
+```python
+primary_account = registry.get_primary()
+```
+
+#### Manage configurations of Spark service accounts
+
+Spark service accounts can have configurations that are provided (unless 
+overridden) during each execution of Spark Jobs. These configuration are abstracted 
+by means of the `PropertyFile` object, that are provided when creating the 
+`ServiceAccount` object, via the `extra_confs` argument (see above). 
+
+The `PropertyFile` object can either be created from a dictionary, as 
+done above
+
+```python
+from spark_client.domain import PropertyFile
+
+static_property = PropertyFile({"my-key": "my-value"})
+```
+
+or also read from a file, e.g. 
+
+```python
+from spark_client.domain import PropertyFile
+
+static_property = PropertyFile.read(defaults.static_conf_file)
+```
+
+`PropertyFile` objects can be merged using the `+` operator
+
+```python
+merged_property = static_property + service_account.extra_confs
+```
+
+And `ServiceAccount` properties can be updated using new "merged" properties 
+via the API provided by the registry
+
+```python
+registry.set_configurations(
+    service_account.id, PropertyFile({"my-key-2": "my-value-2"})
```

Comment:
merged_property instead of PropertyFile({"my-key-2": "my-value-2"}) sounds more continous with the narrative.